src/krb5/_creds.pyx

```diff
@@ -139,6 +151,31 @@ cdef extern from "python_krb5.h":
     ) nogil
 
 
+class TicketFlags(enum.IntFlag):
+    # https://github.com/krb5/krb5-assignments/blob/master/ticket-flags
+    reserved = 1 << 0
+    forwardable = 1 << 1
+    forwarded = 1 << 2
+    proxiable = 1 << 3
+    proxy = 1 << 4
+    may_postdate = 1 << 5
+    postdated = 1 << 6
+    invalid = 1 << 7
+    renewable = 1 << 8
+    initial = 1 << 9
+    pre_authent = 1 << 10
+    hw_authent = 1 << 11
+    transited_policy_checked = 1 << 12
+    ok_as_delegate = 1 << 13
+    enc_pa_rep = 1 << 15
+    anonymous = 1 << 16
+
+    # This is to prevent python >= 3.11 from clearing unknown flags when doing:
+    # flags = flags & ~TicketFlags.forwarded
+    # (Under python 3.11, ~TicketFlags.forwarded will contain only known flags.)
+    _all_flags = (1 << 32) - 1
```

Comment:
Can you share an actual scenario where this is required? When you say `will contain only known flags` do you mean that unsetting the bit will clear out any values that aren't defined here or something else?

Reply:
> Can you share an actual scenario where this is required?

This would be if the KDC sets some flag which is not known to the pykrb5 library and the application wants to see that flag.

> When you say `will contain only known flags` do you mean that unsetting the bit will clear out any values that aren't defined here

Yes.

```
import enum

class MyFlags(enum.IntFlag):
    A = 1
    B = 2

print(int(MyFlags(15)))  # Will print 15
print(int(MyFlags(15) & ~MyFlags.B))  # On python >= 3.11, will print 1, before 3.11 will print 13
print(int(~MyFlags.B))  # On python >= 3.11, will print 1, before then will print -3
```

That means without this on Python >= 3.11, `a & ~TicketFlags.initial` will remove not only the `initial` flag, but also all flags which are not defined in the `TicketFlags` enum.

Reply:
That's definitely surprising behaviour but if that's out of our control here :) Thanks for sharing the details.